telegram/client.py

```diff
@@ -407,17 +425,55 @@ def login(self) -> None:
         while not self._authorized:
             logger.info('[login] current authorization state: %s', authorization_state)
             result = actions[authorization_state]()
+            authorization_state = self._wait_authorization_result(result)
+
+    def _non_blocking_login_routine(self, authorization_state) -> AuthorizationState:
```

Comment:
Since we have authorization states as a part of the library's API, let's merge all these into login method with `block=True` as you suggested. If the `send_code` and `send_password` continue the login process, we definetely don't need the `start_login`. I'm not sure that they should do this, but I do agree that it's more convenient from the user's perspective.

So the `login(block=True)` method will perform all authorization steps and ask codes and passwords using stdin, and `login(block=False)` will return authorization state if it requires any blocking action (code or password in our case).

Reply:
Perfect, I'll merge the two implementations